catalyst_voices/lib/pages/registration/wallet_link/stage/rbac_transaction_panel.dart

```diff
@@ -173,6 +228,57 @@ class _PositiveSmallPrint extends StatelessWidget {
   }
 }
 
+class _BlocTxSubmitError extends StatelessWidget {
+  const _BlocTxSubmitError();
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocWalletLinkBuilder(
+      selector: (state) => state.submittedTx,
+      builder: (context, result) {
+        return switch (result) {
+          Failure(:final value) => _Error(
+              error: value,
+              onRetry: () => _onRetry(context),
+            ),
+          _ => const Offstage(),
+        };
+      },
+    );
+  }
+
+  void _onRetry(BuildContext context) {
+    unawaited(RegistrationCubit.of(context).walletLink.submitRegistration());
+  }
+}
+
+class _Error extends StatelessWidget {
+  final Object error;
+  final VoidCallback onRetry;
+
+  const _Error({
+    required this.error,
+    required this.onRetry,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Align(
+      alignment: Alignment.topCenter,
+      child: Container(
+        padding: const EdgeInsets.only(top: 20),
+        width: double.infinity,
+        child: VoicesErrorIndicator(
+          message: error is WalletApiException
+              ? context.l10n.walletLinkTransactionFailed
+              : context.l10n.somethingWentWrong,
```

Comment:
Maybe for future but we will need some sort of classes for working with Exceptions. Smth. like `LocalizedException`. Not yet sure how it should look like in detail

Reply:
Exceptions should not have localizations themselves, most likely where you create them you don't access to build context that gives you access to localizations. But there should be a way to generically translate an exception into a localization. I'll think about it, maybe even in this PR

Reply:
Ended up with LocalizedException in view_models